Azure/sonic-buildimage
# database-chassis python http-server implementation

Signed-off-by: anamehra [anamehra@cisco.com](mailto:anamehra@cisco.com)
```
  Python HTTP Server service.
  This service runs in database-chassis docker based on the configuration
  provided in device/<platform>/chassisdb.conf

  Configuration parameters provided by chassisdb.conf are as follows:

  start_http_server: if "yes" or "1", http-server will be started on this
      node
  chassis_db_address   : IP address for http-server. This is the same IP used
   for the chassis db server
  http_server_port : Port to bind to, default: 8000
  http_server_dir  : HTTP server home directory path, default: /var/www/

# Example of config:
'''
start_http_server=yes
chassis_db_address=127.0.0.10
http_server_dir=/var/www/tftp
'''
```


#### Why I did it
Enable http-server as database-chassis service on Chassis supervisor to enable LCs to download image and config files from Supervisor.

#### How I did it
Added python3 http-server based implementation in database-chassis as a docker service run by supervisord.

#### How to verify it
Add device/<platform>/chassisdb.conf with the following data:
start_http_server=yes
chassis_db_address=127.0.0.10
http_server_dir=/var/www/tftp

Boot with the image with the above data.

put an image file at /var/www/tftp/filename
check 'systemctl status http-service'
curl http://http-server:8000/filename -o out_file
compare file_out with /var/www/tftp/filename

#### Which release branch to backport (provide reason below if selected)




#### Description for the changelog


#### Ensure to add label/tag for the feature raised. example - PR#2174 under sonic-utilities repo. where, Generic Config and Update feature has been labelled as GCU.

#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
